ericpaulbishop/gargoyle
# DDNS Updates

- Give ddns_updater the ability to pass control to an external "helper script"** for services which require complex updates (i.e. can't be done by a simple GET request to a special URL)
- Update cloudflare service provider to use APIv4 (original API is now deprecated and was not working) using helper script (relies on curl + ca-certificates-combined)
- GUI will warn user if they are trying to setup a provider which relies on a helper script and they do not have it installed
- Fix response matching always triggering Success even if it actually failed

Other services that need to be updated in this way include:
- GoDaddy
- No-IP (maybe? need to test)


Happy to give you access to my API key etc for testing if you want/need.

-------------------------------------
**Helper Scripts will be bash scripts stored in the /usr/lib/ddns-gargoyle/ folder   
Required variables will be set in the config and passed IN ORDER to the script  
The script will always be passed LOCAL_IP, FORCE_UPDATE and VERBOSE as the final 3 variables  
Scripts should be written with this in mind  
Scripts should return:  
- Update failed; exit 3  
- Update not needed; exit 4  
- Update successful; exit 5  
This aligns with ddns updater itself and these are the expected values
